lib/matplotlib/colorbar.py

```diff
@@ -239,28 +239,48 @@ def __init__(self, parent, userax=True):
                 parent._axes.add_child_axes(outer_ax)
                 outer_ax._axes.child_axes.remove(parent)
             else:
-                parent.remove()
+                try:
+                    parent.remove()
+                except ValueError:
+                    pass  # Already removed
         else:
             outer_ax = parent
 
+        # swap axes in the stack if its in there:
+        if outer_ax in fig._localaxes:
+            fig._localaxes.remove(outer_ax)
+            fig._axstack.remove(outer_ax)
+            fig._localaxes.add(self)
+            fig._axstack.add(self)
         inner_ax = outer_ax.inset_axes([0, 0, 1, 1])
         self.__dict__.update(inner_ax.__dict__)
 
         self.outer_ax = outer_ax
         self.inner_ax = inner_ax
+
         self.outer_ax.xaxis.set_visible(False)
         self.outer_ax.yaxis.set_visible(False)
         self.outer_ax.set_facecolor('none')
         self.outer_ax.tick_params = self.inner_ax.tick_params
         self.outer_ax.set_xticks = self.inner_ax.set_xticks
         self.outer_ax.set_yticks = self.inner_ax.set_yticks
-        for attr in ["get_position", "set_position", "set_aspect"]:
+        for attr in ["get_position", "set_aspect",
+                     "_remove_method", "_set_position",
+                     "set_position", "cla", "draw"]:
```

Comment:
I'm still not sure about these only being mapped to the outer_ax... I would have thought it should be something like:

```python
def draw(self, renderer):
    super().draw(renderer)
    self.outer_ax.draw(renderer)
```

so that you draw _both_ the inner and outer? Same for the cla and remove. The position's I think are appropriate to map straight to the outer though.

Reply:
inner is an artist on outer, so inner gets drawn if outer gets drawn ;-)

Reply:
Oh yeah, it is an inset of the outer! Makes sense now, thanks for clarifying.